File: VB/WinSolution.Module/MyRole.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.Xpo

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace WinSolution.Module
	<DefaultClassOptions> _
	Public Class MyRole
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _roleName As String
		Public Property RoleName() As String
			Get
				Return _roleName
			End Get
			Set(ByVal value As String)
				SetPropertyValue("RoleName", _roleName, value)
			End Set
		End Property
		<Association("MyRoles-MyUsers")> _
		Public ReadOnly Property MyUsers() As XPCollection(Of MyUser)
			Get
				Return GetCollection(Of MyUser)("MyUsers")
			End Get
		End Property
	End Class

End Namespace

File: CS/WinSolution.Module/MyRole.cs
using System;

using DevExpress.Xpo;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace WinSolution.Module {
    [DefaultClassOptions]
    public class MyRole : BaseObject {
        public MyRole(Session session) : base(session) { }
        private string _roleName;
        public string RoleName {
            get { return _roleName; }
            set { SetPropertyValue("RoleName", ref _roleName, value); }
        }
        [Association("MyRoles-MyUsers")]
        public XPCollection<MyUser> MyUsers {
            get {
                return GetCollection<MyUser>("MyUsers");
            }
        }
    }

}
